geowebcache/core/src/test/java/org/geowebcache/config/DefaultingConfigurationTest.java

```diff
@@ -0,0 +1,155 @@
+package org.geowebcache.config;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.geowebcache.GeoWebCacheException;
+import org.geowebcache.grid.GridSetBroker;
+import org.geowebcache.layer.TileLayer;
+import org.geowebcache.layer.wms.WMSHttpHelper;
+import org.geowebcache.layer.wms.WMSLayer;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Set;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.junit.Assert.assertNull;
+public class DefaultingConfigurationTest {
+
+    DefaultingConfiguration config = getConfig();
+    TileLayer tl = new WMSLayer("test", new String[] {"http://example.com/"}, null,
+            Integer.toString(1),null, null, null, null,
+            null, false, null);
+    GeoWebCacheConfiguration gwcConfig = new GeoWebCacheConfiguration();
+    GridSetBroker gridSetBroker =  new GridSetBroker(true, true);
+
+    private static Log log = LogFactory.getLog(org.geowebcache.config.XMLConfiguration.class);
+
+    @Before
+    public void setUpTestUnit() throws Exception {
+        config.setDefaultValues(tl);
```

Comment:
It may be useful to do some verification that values are not set (or alternatively, manually unset the values in question) before calling `config.setDefaultValues`, so that you test that the DefaultingConfiguration actually does something (actually changes the values).

Reply:
Added test on new tile layer with unset configurations, and refactored the other tests.